org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTags.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2014 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.tasks;
+
+import static java.util.stream.Collectors.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * @author Stefan Oehme - Initial contribution and API
+ * @since 2.6
+ */
+public class TaskTags implements Iterable<TaskTag> {
+	private boolean caseSensitive;
+
+	private final List<TaskTag> taskTags = new ArrayList<>();
+
+	private Pattern pattern;
+
+	private Map<String, TaskTag> taskTagsByName;
+
+	@Override
+	public Iterator<TaskTag> iterator() {
+		return taskTags.iterator();
+	}
+
+	public Pattern toPattern() {
+		if (pattern == null) {
+			int flags = Pattern.MULTILINE;
+			if (!caseSensitive) {
+				flags = flags | Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE;
+			}
+			StringBuilder builder = new StringBuilder();
+			builder.append("^.*((");
+			String tagNames = taskTags.stream().map(it -> Pattern.quote(it.getName())).collect(joining("|"));
```

Comment:
could use `collect(joining(delimiter, prefix, suffix))` and simplify the code.

Reply:
This is still unresolved, isn't it?

Reply:
Actually I wasn't sure what you meant here. collect(joining()) is already used. How would you simplify it?

Reply:
The code builds a new StringBuilder, appends a prefix, uses joining and appends a suffix. joining is overloaded and accepts a prefix and suffix along with the delimiter, too.
You can create the entire string in one shot.